File: ui/src/app/models/circle.model.ts
// Models matching backend Circle API structure

export interface Circle {
  circleResourceId: string;
  name: string;
  description?: string;
  ownerSubjectId: string;
  isOwner?: boolean;
  memberCount?: number;
  recipeCount?: number;
  createdDate?: string;
}

export interface CircleMember {
  circleMemberResourceId?: string;
  circleId?: string;
  subjectId: string;
  userEmail?: string;
  role: CircleRole;
  status: MemberStatus;
  joinedDate?: string;
  invitedDate?: string;
}

export enum CircleRole {
  Owner = 'Owner',
  Admin = 'Admin',
  Member = 'Member'
}

export enum MemberStatus {
  Pending = 'Pending',
  Active = 'Active',
  Left = 'Left'
}

export interface CircleRecipe {
  circleRecipeResourceId?: string;
  circleId?: string;
  recipeResourceId: string;
  title?: string;
  recipeImageUrl?: string;
  sharedBySubjectId: string;
  sharedByEmail?: string;
  sharedDate: string;
}

export interface CircleInvite {
  circleInviteResourceId?: string;
  inviteToken: string;
  circleId?: string;
  circleName?: string;
  inviteeEmail?: string;
  invitedBySubjectId?: string;
  invitedByEmail?: string;
  createdDate?: string;
  expiresDate: string;
  status: InviteStatus;
}

export enum InviteStatus {
  Pending = 'Pending',
  Accepted = 'Accepted',
  Expired = 'Expired',
  Revoked = 'Revoked'
}

// Request/Response DTOs

export interface CreateCircleRequest {
  name: string;
  description?: string;
}

export interface UpdateCircleRequest {
  name?: string;
  description?: string;
}

export interface InviteMemberRequest {
  inviteeEmail?: string;
  expiresDate?: string;
}

export interface ShareRecipeRequest {
  recipeResourceId: string;
}

export interface AcceptInviteRequest {
  inviteToken: string;
}

export interface PagedResult<T> {
  items: T[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}

File: src/RecipeVault.WebApi/Models/Responses/CircleModel.cs
#pragma warning disable CS1591 // Missing XML comments

using System;
using System.Collections.Generic;

namespace RecipeVault.WebApi.Models.Responses {
    public class CircleModel {
        public Guid CircleResourceId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public Guid OwnerSubjectId { get; set; }
        public DateTime CreatedDate { get; set; }
        public List<CircleMemberModel> Members { get; set; }
        public List<CircleRecipeModel> SharedRecipes { get; set; }
        public int MemberCount { get; set; }
        public int RecipeCount { get; set; }
        public bool IsOwner { get; set; }
    }

    public class CircleMemberModel {
        public Guid SubjectId { get; set; }
        public string Role { get; set; }
        public string Status { get; set; }
        public DateTime? JoinedDate { get; set; }
    }

    public class CircleRecipeModel {
        public Guid RecipeResourceId { get; set; }
        public string Title { get; set; }
        public Guid SharedBySubjectId { get; set; }
        public DateTime SharedDate { get; set; }
    }

    public class CircleInviteModel {
        public Guid InviteToken { get; set; }
        public string CircleName { get; set; }
        public string InviteeEmail { get; set; }
        public DateTime ExpiresDate { get; set; }
        public string Status { get; set; }
    }
}
